### PR TITLE
doc: add RIOT overview graphic

### PR DESCRIPTION
**Pull request by authmillenon**
This introduces the graphic introduced in https://github.com/RIOT-OS/RIOT/issues/4072#issuecomment-147752215 to the repository. The SVG was designed in Inkscape, but I took great care to clean-up all unnecessary garbage added by Inkscape and hand-edited the SVG for easy readability (I condensed the size down to ~300 lines of XML code).

Depends on #4091.

### CODE REVIEW

File: doc/doxygen/src/riot-overview.svg [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+  <metadata>
+    <rdf:RDF>
+      <cc:Work rdf:about="Provides a general overview over RIOT's structure">
+        <dc:format>image/svg+xml</dc:format>
+        <dc:type rdf:resource="http://purl.org/dc/dcmitype/StillImage" />
+        <dc:title>RIOT Overview</dc:title>
+        <cc:license rdf:resource="http://creativecommons.org/licenses/by-sa/3.0/" />
+        <dc:creator>
+          <cc:Agent>
+            <dc:title>Cenk Gündogan</dc:title>
```
Comment (authmillenon) [inline doc/doxygen/src/riot-overview.svg]: @cgundogan I wasn't sure where the special characters go in your name. Please check.

Comment (cgundogan) [inline doc/doxygen/src/riot-overview.svg]: Almost correct: Gündoğan
Am 13.10.2015 23:43 schrieb "Martine Lenders" <notifications@github.com>:

Comment (authmillenon) [inline doc/doxygen/src/riot-overview.svg]: fixed :)

Comment (cgundogan) [inline doc/doxygen/src/riot-overview.svg]: Thanks (:
Am 13.10.2015 23:49 schrieb "Martine Lenders" <notifications@github.com>:

**Comment by OlegHahm** [comment]
Including nano and lwip without having it already in master seems confusing to me. Otherwise I like the diagram very much. Good work!

**Comment by authmillenon** [comment]
True, but lwIP is about to be added and nano can always be replaced with CCNlite or the likes. I just wanted to give an impression of RIOT being not dependent on the stack, while presenting `gnrc` as the preferred way to go.

**Comment by authmillenon** [comment]
Reversed dependency to #4091 (this PR now depends on it).

**Comment by authmillenon** [comment]
(Rebased to current master)

**Comment by authmillenon** [comment]
ping?

**Comment by Kijewski** [comment]
> Sorry, this file is invalid so it cannot be displayed.

https://github.com/authmillenon/RIOT/blob/doc/enh/overview-graphic/doc/doxygen/src/miso.svg

**Comment by authmillenon** [comment]
It's a font. It's not supposed to be displayed on its own. It is displayed correctly in the the image SVGs.

**Comment by emmanuelsearch** [comment]
ACK for the structure part.

**Comment by authmillenon** [comment]
Fixed the whitespace issue, removed the original graphic and updated the commit names accordingly.

**Comment by emmanuelsearch** [comment]
ACK

**Pull request merged**